### PR TITLE
Mcumgr revision update

### PR DESCRIPTION
**Pull request by de-nordic**
The mcumgr update from upstream with needed smp_svr configuration changes and documentation update.

### CODE REVIEW

**Comment by de-nordic** [comment]
DNM because requires testing and merging in modules.

File: samples/subsys/mgmt/mcumgr/smp_svr/README.rst [hunk 1]
```diff
@@ -263,6 +263,39 @@ The general sequence of a DFU process is as follows:
 * Reset the device remotely using :file:`mcumgr`
 * Confirm the uploaded image using :file:`mcumgr` (optional)
 
+Slot to image mapping and selection
+===================================
+
+Currently the mcumgr supports 4 images of which first two are mapped into
+primary and secondary slot. Default upload is performed to "image-1", which
+is slot-0, and the application is always running from "image-0", that is slot-1.
```
Comment (nvlsianpu) [inline samples/subsys/mgmt/mcumgr/smp_svr/README.rst:271]: in table below image-1 is same row as slot-1, similar image-0 and slot-0
I don't get what this sentence mean.

Comment (de-nordic) [reply samples/subsys/mgmt/mcumgr/smp_svr/README.rst:271]: Ha, because it is wrong.

**Comment by de-nordic** [comment]
@aunsbjerg @carlescufi @nashif @nvlsianpu please review.

**Comment by kurddt** [comment]
@de-nordic Will this change allow to update using swapping two dependent image. Or does MCUBoot ignore the content of `image-2` and `image-3` ?

**Comment by de-nordic** [comment]
> 
> 
> @de-nordic Will this change allow to update using swapping two dependent image. Or does MCUBoot ignore the content of `image-2` and `image-3` ?

No it will not. There is `MCUBOOT_IMAGE_NUMBER==2` that allows to use two pairs of slots (Zephyr images) to boot, but the additions you are mentioning here are for use with specific scenario where you have two cores that will be obtaining update but only one core is able to communicate to mcumgr. In such case the two other images can be mapped for the other mcu/core.

**Review by mbolivar-nordic — APPROVED** [review]

**Review by aunsbjerg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**